src/main/java/com/google/sps/placeGuide/repository/impl/DatastorePlaceGuideRepository.java

```diff
@@ -0,0 +1,121 @@
+
+    return placeGuideEntity;
+  }
+
+  @Nullable
+  @Override
+  public List<PlaceGuide> getAllPlaceGuides() {
+    Filter queryFilter = new FilterPredicate(IS_PUBLIC_PROPERTY, FilterOperator.EQUAL, true);
+    Query query = new Query(ENTITY_KIND).setFilter(queryFilter);
+    return getPlaceGuidesList(query);
+  }
+
+  @Nullable
+  @Override
+  public List<PlaceGuide> getCreatedPlaceGuides(String creatorId) {
+    Filter queryFilter = new FilterPredicate(CREATOR_ID_PROPERTY, FilterOperator.EQUAL, creatorId);
+    Query query = new Query(ENTITY_KIND).setFilter(queryFilter);
+    return getPlaceGuidesList(query);
+  }
+
+  @Nullable
+  @Override
+  public List<PlaceGuide> getCreatedPublicPlaceGuides(String creatorId) {
+    Filter queryFilter = CompositeFilterOperator.and(Arrays.asList(
+                        new FilterPredicate(CREATOR_ID_PROPERTY, FilterOperator.EQUAL, creatorId),
+                        new FilterPredicate(IS_PUBLIC_PROPERTY, FilterOperator.EQUAL, true)));
+    Query query = new Query(ENTITY_KIND).setFilter(queryFilter);
+    return getPlaceGuidesList(query);
+  }
+
+  @Nullable
+  @Override
+  public List<PlaceGuide> getCreatedPrivatePlaceGuides(String creatorId) {
+    Filter queryFilter = CompositeFilterOperator.and(Arrays.asList(
+                        new FilterPredicate(CREATOR_ID_PROPERTY, FilterOperator.EQUAL, creatorId),
+                        new FilterPredicate(IS_PUBLIC_PROPERTY, FilterOperator.EQUAL, false)));
+    Query query = new Query(ENTITY_KIND).setFilter(queryFilter);
+    return getPlaceGuidesList(query);
+  }
+
+  @Nullable 
+  private List<PlaceGuide> getPlaceGuidesList(Query query) {
+    PreparedQuery results = datastore.prepare(query);
+    List<PlaceGuide> createdPlaceGuides = new ArrayList<>();
+    for (Entity placeGuideEntity : results.asIterable()) {
+      long id = placeGuideEntity.getKey().getId();
+      String name = (String) placeGuideEntity.getProperty(NAME_PROPERTY);
+      String audioKey = (String) placeGuideEntity.getProperty(AUDIO_KEY_PROPERTY);
+      String creatorId = (String) placeGuideEntity.getProperty(CREATOR_ID_PROPERTY);
+      String placeId = (String) placeGuideEntity.getProperty(PLACE_ID_PROPERTY);
+      GeoPt coord = (GeoPt) placeGuideEntity.getProperty(COORD_PROPERTY);
+      boolean isPublic = (boolean) placeGuideEntity.getProperty(IS_PUBLIC_PROPERTY);
+      String desc = (String) placeGuideEntity.getProperty(DESC_PROPERTY);
+      long length = (long) placeGuideEntity.getProperty(LENGTH_PROPERTY);
+      String imgKey = (String) placeGuideEntity.getProperty(IMG_KEY_PROPERTY);
+      PlaceGuide placeGuide = new PlaceGuide
+                              .Builder(name, audioKey, creatorId, placeId, coord)
+                              .setId(id).setLength(length).setDescription(desc)
+                              .setImageKey(imgKey).setPlaceGuideToPublic(isPublic)
+                              .build();
```

Comment:
EXtract a function to create a PlaceGudie from placeGuideEntity.

Reply:
done.